crates/service_utils/src/service/types.rs

```diff
@@ -228,3 +240,31 @@ impl FromRequest for CustomHeaders {
         ready(Ok(val))
     }
 }
+
+pub struct CustomResp<T> {
+    pub body: T,
+    pub headers: HashMap<String, String>,
+}
+impl<T: Serialize> Responder for CustomResp<T> {
+    type Body = EitherBody<String>;
+
+    fn respond_to(self, _: &HttpRequest) -> HttpResponse<Self::Body> {
+        match (serde_json::to_string(&self.body), self.headers) {
```

Comment:
Is there any benefit of doing this? We still end up writing the code to add headers in every function. Why can't I just add the headers as well? I don't need the responder for that.